supertokens_python/recipe/session/cookie_and_header.py

```diff
@@ -13,43 +13,49 @@
 # under the License.
 from __future__ import annotations
 
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, List, Optional
 from urllib.parse import quote, unquote
 
 from typing_extensions import Literal
 
 from .constants import (
     ACCESS_CONTROL_EXPOSE_HEADERS,
     ACCESS_TOKEN_COOKIE_KEY,
+    ACCESS_TOKEN_HEADER_KEY,
     ANTI_CSRF_HEADER_KEY,
+    AUTH_MODE_HEADER_KEY,
+    AUTHORIZATION_HEADER_KEY,
     FRONT_TOKEN_HEADER_SET_KEY,
-    ID_REFRESH_TOKEN_COOKIE_KEY,
-    ID_REFRESH_TOKEN_HEADER_SET_KEY,
     REFRESH_TOKEN_COOKIE_KEY,
+    REFRESH_TOKEN_HEADER_KEY,
     RID_HEADER_KEY,
+    available_token_transfer_methods,
 )
 
 if TYPE_CHECKING:
     from supertokens_python.framework.request import BaseRequest
     from supertokens_python.framework.response import BaseResponse
     from .recipe import SessionRecipe
+    from .utils import TokenTransferMethod, TokenType, SessionConfig
 
 from json import dumps
 from typing import Any, Dict, Union
 
-from supertokens_python.exceptions import raise_general_exception
 from supertokens_python.utils import get_header, utf_base64encode
 
 
 def set_front_token_in_headers(
-    response: BaseResponse,
+    response: Optional[BaseResponse],
```

Comment:
`Optional` doesn't mean the param will be optional. It just means that it will accept `None` as well. 

We are allowing it to pass `None` so that we can add function calls to `response_mutators` and later on pass the actual response object later on in the `mange_session_post_response` function.

Reply:
I see that you've removed Optional now, so I'll just resolve this comment. The reason I asked about this is that as far as I could tell adding these calls to `response_mutators` with `partial` worked even without allowing null.